python/tests/conftest.py

```diff
@@ -0,0 +1,11 @@
+import ctypes
+
+import pytest
+
+def pytest_addoption(parser):
+    parser.addoption("--lib", action="store", default="libfixturecsm")
+
+@pytest.fixture(scope='module')
+def loadlib(request):
+    libname =  request.config.getoption("--lib")
+    return ctypes.CDLL(libname)
```

Comment:
Create a fixture with the arg passed via --lib. Tests then use this fixture. The fixture is loaded once per module (not ideal, but the package level fixture is in alpha in pytest).